Include/Rover/ListTrial.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef ROVER_LIST_TRIAL_HPP
+#define ROVER_LIST_TRIAL_HPP
+#include <vector>
+
+namespace Rover {
+
+  //! Stores samples as an array in RAM
+  /*!
+    \tparam S The type of the samples.
+  */
+  template<typename S>
+  class ListTrial {
+    public:
+      
+      //! The type of stored samples.
+      using Sample = S;
+
+      //! The type of the constant iterator.
+      using ConstIterator = typename std::vector<Sample>::const_iterator;
+  
+      //! Increases the capacity of the underlying array.
+      void reserve(std::size_t capacity);
+
+      //! Inserts a sample to this trial.
+      void insert(const Sample& s);
+  
+      //! Inserts a sample to this trial.
+      void insert(Sample&& s);
+
+      //! Inserts all samples from another trial to this one.
```

Comment:
Remove the ```ListTrial``` overloads, these will be part of a more general free function that can work for any ```Trial```.